### PR TITLE
Move `ruff-lsp` and fix its wrapper

### PR DESCRIPTION
**Pull request by CobaltCause**
###### Description of changes

This makes it possible to use `ruff-lsp` in Python environments that contain incompatible versions of Python dependencies required by `ruff-lsp`.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/language-servers/ruff-lsp/default.nix [hunk 1]
```diff
@@ -53,6 +53,11 @@ buildPythonPackage rec {
   makeWrapperArgs = [
     # prefer ruff from user's PATH, that's usually desired behavior
     "--suffix PATH : ${lib.makeBinPath [ ruff ]}"
+
+    # Unset ambient PYTHONPATH in the wrapper, so Poetry only ever runs with
```
Comment (SuperSandro2000) [inline pkgs/development/tools/language-servers/ruff-lsp/default.nix:57]: I think poetry is wrong here

**Review by SuperSandro2000 — COMMENTED** [review]
please add an alias to python-aliases.nix

File: pkgs/development/tools/language-servers/ruff-lsp/default.nix (hunk 1 above)
Comment (CobaltCause) [reply pkgs/development/tools/language-servers/ruff-lsp/default.nix:57]: Oh yes, I just copypasted the comment from its file lol.

File: pkgs/top-level/python-aliases.nix [hunk 2]
```diff
@@ -291,6 +291,7 @@ mapAliases ({
   ruamel_base = ruamel-base; # added 2021-11-01
   ruamel_yaml = ruamel-yaml; # added 2021-11-01
   ruamel_yaml_clib = ruamel-yaml-clib; # added 2021-11-01
+  ruff-lsp = throw "ruff-lsp was promoted to a top-level attribute"; # added 2023-06-23
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-aliases.nix:294]:
```suggestion
  inherit (super.pkgs) ruff-lsp; # added 2023-06-23
```

File: pkgs/development/tools/language-servers/ruff-lsp/default.nix [hunk 3]
```diff
@@ -53,6 +53,13 @@ buildPythonPackage rec {
   makeWrapperArgs = [
     # prefer ruff from user's PATH, that's usually desired behavior
     "--suffix PATH : ${lib.makeBinPath [ ruff ]}"
+
+    # Stolen from `pkgs.poetry`:
+    #
```
Comment (SuperSandro2000) [inline pkgs/development/tools/language-servers/ruff-lsp/default.nix:58]:
```suggestion
```

File: pkgs/top-level/python-aliases.nix (hunk 2 above)
Comment (CobaltCause) [reply pkgs/top-level/python-aliases.nix]: Oh neat

**Review by figsoda — APPROVED** [review]

**Pull request merged**